airflow/utils/email.py

```diff
@@ -267,8 +306,13 @@ def _get_smtp_connection(host: str, port: int, timeout: int, with_ssl: bool) ->
 
 
 def _get_email_list_from_str(addresses: str) -> list[str]:
-    delimiters = [",", ";"]
-    for delimiter in delimiters:
-        if delimiter in addresses:
-            return [address.strip() for address in addresses.split(delimiter)]
-    return [addresses]
+    """
+    Extract a list of email addresses from a string. The string
+    can contain multiple email addresses separated by
+    any of the following delimiters: ',' or ';'.
+
+    :param addresses: A string containing one or more email addresses.
+    :return: A list of email addresses.
+    """
+    pattern = r"[,;]\s*"
+    return [address.strip() for address in re.split(pattern, addresses)]
```

Comment:
Since we do `strip` afterwards, the `\s*` part in the pattern does not seem to be needed? (Wildcards incur a minor performance penalty.)

An alternative is to do `\s*` at both sides and remove the extra `strip`.